### PR TITLE
add support for 'format' command line parameter

### PR DESCRIPTION
**Pull request by egirshov**
Pass through 'format' parameter which is handled by node-gyp configure step
(https://github.com/nodejs/node-gyp/blob/master/lib/configure.js#L216)

### CODE REVIEW

**Comment by ralphtheninja** [comment]
@egirshov Sorry for not responding in a timely fashion. Mind rebasing this onto latest master? Lets get this merged!

**Comment by ralphtheninja** [comment]
@egirshov Remind me again, what the `--format` is used for? I'm looking through the `node-gyp` code to figure it out and at a quick glance it seems related to which system you use for building, e.g. `make` or `msvs` for windows.

What is your use case? Just curious.

**Comment by egirshov** [comment]
I noticed that `node-gyp` supports composite formats (i.e. `format-flavor`), see https://github.com/nodejs/node-gyp/blob/master/gyp/pylib/gyp/__init__.py#L62 while by default the format is either `msvs` or `make` (https://github.com/nodejs/node-gyp/blob/master/lib/configure.js#L216).
In particular, I needed to set the format to `make-android` in order to force cross-compilation mode when building leveldown (the pull request is referenced above, I'll rebase it as well as some point).

I guess having the choice of build system might be useful in other case as well.

File: gypbuild.js [hunk 1]
```diff
@@ -28,6 +28,8 @@ function runGyp (opts, target, cb) {
     }
     if (opts.debug) args.push('--debug')
 
+    if (opts.format) args.push('--', '--format', opts.format)
```
Comment (ralphtheninja) [inline gypbuild.js]: Why is `--` needed here? Isn't `--format` part of `node-gyp`'s arguments?

**Comment by ralphtheninja** [comment]
@egirshov Cool! It would be nice if you could add some docs on `--format` with a simple example, perhaps mention the `make-android` use case specifically.

File: gypbuild.js (hunk 1 above)
Comment (ralphtheninja) [reply gypbuild.js]: I'm guessing `node-gyp` will take the arguments after `--` and pass them directly to the python code. I just don't see any docs on it.

Comment (egirshov) [reply gypbuild.js]: The logic in `parseArgv` is somewhat convoluted (https://github.com/nodejs/node-gyp/blob/master/lib/node-gyp.js#L135-L147), as far as I understood, the regular arguments (those before `--`) are consumed in that function while after `--` are passed to the python code.

**Comment by egirshov** [comment]
updated readme (added example)

**Comment by ralphtheninja** [comment]
7.4.0

**Pull request merged**